htdocs/core/class/hookmanager.class.php

```diff
@@ -251,6 +253,7 @@ function executeHooks($method, $parameters=array(), &$object='', &$action='')
 
                     	//dol_syslog("Call method ".$method." of class ".get_class($actionclassinstance).", module=".$module.", hooktype=".$hooktype, LOG_DEBUG);
                     	$resaction = $actionclassinstance->$method($parameters, $object, $action, $this); // $object and $action can be changed by method ($object->id during creation for example or $action to go back to other action for example)
+						$this->resPrint=''; $this->resArray=array(); // Init return properties
```

Comment:
If we call 3 hooks at same level, we want result of hook to be concatenated. For example 3 hooks, at same level, that add html content return into ->resprints
And we want to concat the 3 into $resPrint so once the 3 pass into the loop of $actionclassinstance is finished, we have a ->resPrint that is complete. But here you empty the result obtained by previous call.

Reply:
we work on it